x-pack/performance/journeys/aiops_log_rate_analysis.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { Journey } from '@kbn/journeys';
+import { subj } from '@kbn/test-subj-selector';
+import type { DataView } from '@kbn/data-views-plugin/common';
+
+// We are importing from tests to tests here, not related to runtime code.
+// eslint-disable-next-line @kbn/imports/no_boundary_crossing
+import { frequentItemSetsLargeArraysSource } from '../../test/functional/apps/aiops/log_rate_analysis/test_data/__mocks__/frequent_item_sets_large_arrays';
+
+const indexName = 'large_arrays';
+let dataViewId: string | undefined;
+
+interface DocWithArray {
+  '@timestamp': number;
+  items: string[];
+}
+
+export const journey = new Journey({
+  beforeSteps: async ({ kbnUrl, log, auth, es, kibanaServer }) => {
+    // Create index with mapping
+    try {
+      await es.indices.create({
+        index: indexName,
+        mappings: {
+          properties: {
+            items: { type: 'keyword' },
+            '@timestamp': { type: 'date' },
+          },
+        },
+      });
+    } catch (e) {
+      log.error(`Error creating index '${indexName}' in beforeSteps() callback`);
+    }
+
+    // Ingest sample data
+    try {
+      await es.bulk({
+        refresh: 'wait_for',
+        body: frequentItemSetsLargeArraysSource.reduce((docs, items) => {
+          if (docs === undefined) return [];
+          docs.push({ index: { _index: indexName } });
+          docs.push({
+            '@timestamp': 1562254538700,
+            items,
+          });
+          return docs;
+        }, [] as estypes.BulkRequest<DocWithArray, DocWithArray>['body']),
+      });
+    } catch (e) {
+      log.error(`Error ingesting sample data into '${indexName}' in beforeSteps() callback`);
+    }
+
+    // Create data view
+    try {
+      const dataViewResp = await kibanaServer.request<DataView>({
+        method: 'POST',
+        path: '/api/saved_objects/index-pattern',
+        body: {
+          attributes: { title: indexName, timeFieldName: '@timestamp' },
+        },
+      });
+
+      dataViewId = dataViewResp.data.id;
+    } catch (e) {
+      log.error(`Error creating data view for index '${indexName}' in beforeSteps() callback`);
+    }
+  },
```

Comment:
Data view has to be re-created because we start Kibana for warmup and again for test phase.